### PR TITLE
AccessibleObject NRT annotations

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

Annotate the following types:
* AccessibleObject
* InternalAccessibleObject
* AccessibleObject.EnumVariantObject
* Control.ControlAccessibleObject

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3273)

### CODE REVIEW

**Comment by RussKie** [comment]
📢  @M-Lipin @Ryuugamine @vladimir-krestov you will need this as a baseline for all your a11y work

File: src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs [hunk 1]
```diff
@@ -1062,26 +1049,30 @@ object IAccessible.accNavigate(int navDir, object childID)
...
-                AccessibleObject child = GetAccessibleChild(childID);
+                AccessibleObject? child = GetAccessibleChild(childID);
                 if (child != null)
                 {
-                    return AsVariant(child.Navigate((AccessibleNavigation)navDir));
+                    AccessibleObject? newObject = child.Navigate((AccessibleNavigation)navDir);
+                    if (newObject != null)
+                    {
+                        return AsVariant(newObject);
+                    }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs:1067]: @M-Lipin this change can have side effects, i.e. if `newObject == null` we can now fall through to `if (systemIAccessible != null)` condition.
Is this acceptable, or should we return `null` here?

**Comment by RussKie** [comment]
Waiting for @M-Lipin to review, if no concerns or objections I will fix the tests.

File: src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs (hunk 1 above)
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs]: @M-Lipin please raise a follow up issue to review the correctness of the implementation.

Comment (M-Lipin) [reply src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs]: It seems to me that we should not change the implementation at this moment as `AsVariant` method has inner check for null so it is allowed to pass null as its parameter. I have filed a separate issue to review this specific statement separately #3287
So please keep this as it was before to not accidentally bring the functionality change with this PR.
I've also filed an issue to implement tests to cover accessible navigation, so we'll be able to review the functional result of existing code flows:  #3285

**Pull request merged**